lasio/reader.py

```diff
@@ -64,6 +71,9 @@ def open_file(file_ref, **encoding_kwargs):
         was used to decode it (if it were read from disk).
 
     """
+
+    file_ref = check_for_path_obj(file_ref)
```

Comment:
Thank you @dcslagel for giving this a go. Will be nice to have pathlib.Path support!

We may need to incorporate this into the logic following line 78. The reason is that, where it is now, it will miss all the encoding sniffing goodness that lasio performs via the `open_with_codecs` function.

Reply:
Hi @dagrha, 

okay, change 9f45d2a adds pathlib tests to test_encoding.py to check for the the encoding of pathlib.Path objects.  

Then to make sure the Path objs go through the encoding analysis,  check_for_path_obj() now returns the full path string filename of the object.  This string should then get processed properly by 'def instance(file_ref, str):...'. 

The pathlib tests all pass and match the encodings for same las files as used in the non-pathlib tests.

See if this is a good solution or needs some additional tuning.

Thanks!,

DC